### ISSUE TITLE
Hi Cadene, is the Tucker decomposition part in fusion.py?

### ISSUE THREAD

**Issue by ghost** [Cadene/vqa.pytorch#31]
Hi Cadene,

I think the decomposition part is in fusion.py. But i have get them all clear yet. Would mind helping me out? Because I want to code a Tensorflow version based on your pytorch code. And I also find that the BayesianUniSkip(skipthoughts) part is a challenge part for me, because i also didn't find the Tensorflow version. Thanks for you kind help.

**Comment by Cadene** [comment]
Edit: Candene -> Cadene

Better now :D

> I think the Tucker decomposition part is in fusion.py.

No, it is not. Unfortunately, we don't provide a pytorch implementation of the Tucker Decomposition, but only of Mutan (Tucker Decomposition with rank constraint). We plan to add it in few weeks (using nn.Bilinear).

> And I also find that the BayesianUniSkip(skipthoughts) part is a challenge part for me, because i also didn't find the Tensorflow version.

It was very difficult for us because the original implementation is in Theano. Nevertheless, you should be able to port it in tensorflow using these classes which do not rely too much on a pytorch specific implementation:
- https://github.com/Cadene/skip-thoughts.torch/blob/master/pytorch/gru.py
- https://github.com/Cadene/skip-thoughts.torch/blob/master/pytorch/skipthoughts.py
Unfortunately having a pretrained Embedding + RNN with Sequential Dropout (BayesianGRU) turned to be critical for us (1 to 2% gain in accuracy).
